### PR TITLE
Convert to pip module

### PR DESCRIPTION
**Pull request by BenBrostoff**
This PR will address #63 . [I've published a version (`1.0.0`) to pip](https://pypi.org/project/draftfast/). I'm putting my TODO list here so people can follow along as this gets developed:

- [x] - Remove all `Namespace` / CLI stuff
- [x] - Expose rules
- [x] - Merge with master once @sharkiteuthis 's PR is merged
- [x] - Add repl.it sandbox link to README

Not completed, okay to move to issues:
- [ ] - Allow option to control flex position (blacklist or whitelist)
- [ ] - Allow user to pass in custom functions for player pool

@sharkiteuthis - ~~this isn't ready yet but I'm hoping to finish in the next few days. If you're working on other branches it may be useful to periodically pull from this branch.~~ I've released `1.0.0` and even though this is a little rough around the edges, I think we should merge to master so working on feature branches is easier. The three things that are core functionality that are not in the exposed library right now are 1) uploading CSVs to DraftKings / FanDuel and 2) exposure work you completed (it's there actually but not documented or tested) and 3) pick em optimizer for DraftKings. I can create issues for all three.

I invited you to the repo so you can have review access - let me know if you didn't get it 👍 

UPDATE - [you can experiment with the exposed module on repl.it here](https://repl.it/@BenBrostoff/AllWarlikeDemoware)

### CODE REVIEW

**Comment by sharkiteuthis** [comment]
let me know when you want me to review

**Comment by BenBrostoff** [comment]
Will do, this might take a few days because the API is changing in a significant way.

**Comment by sharkiteuthis** [comment]
What are these items?

> - Expose rules
> - Allow option to control flex position (blacklist or whitelist)
> - Allow user to pass in custom functions for player pool

Do you mind explaining what you have in mind here? Is there overlap with #94?

**Comment by BenBrostoff** [comment]
Ah that’s my fault, I didn’t see 94. The flex is overlap but the rules just
refer to objects that contain data on salary cap / roster size / etc based
on the sport and site. The custom function is a way to filter the players
by some arbitrary condition like has an average of at least 10 points or is
on the Clippers but is not a PG.

**Comment by sharkiteuthis** [comment]
I like the rules idea. In general I don't stack QB/RB unless it's Dak/Zeke, Jackson/Edwards, or Cam/CMC. It would be cool to handle that with a single rule instead of having to write 29 groups

File: README.md [hunk 1]
```diff
@@ -1,97 +1,118 @@
...
+from draftfast import rules
 
-```
-python optimize.py --help
+nhl_rules = rules.RuleSet(
+    site=rules.DRAFT_KINGS,
+    league='NHL',
+    roster_size='9',
+    position_limits=[['C', 9, 9]],
+    salary_max=50_000,
```
Comment (sharkiteuthis) [inline README.md]: I get a syntax error here

File: draftfast/rules.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        'NBA': 9,
+        'MLB': 9,
+        'WNBA': 7,
+        'NASCAR': 5,
+        'PGA': 6,
+    }
+}
+
+
+SALARY_CAP = {
```
Comment (sharkiteuthis) [inline draftfast/rules.py]: and here

**Comment by BenBrostoff** [comment]
@sharkiteuthis Ugh, this is a Python 3.5 thing. I'm upgrading the `Dockerfile` to Ubuntu 17.10 which I think uses 3.6 by default, will see if that resolves.

**Comment by BenBrostoff** [comment]
Also the Travis build is no longer triggering. I think this is related to them deprecating travis-ci.org... not worrying about it for now.

**Comment by BenBrostoff** [comment]
Okay @sharkiteuthis , once you approve this I'll merge in 👍 . Added the issues you mentioned.

**Comment by sharkiteuthis** [comment]
building the new docker image, numpy takes forever on my laptop

**Comment by sharkiteuthis** [comment]
works for me

**Pull request merged**